### ISSUE TITLE
Zero reward issue

### ISSUE THREAD

**Issue by daochenzha** [facebookresearch/adversarially-motivated-intrinsic-goals#1]
Hi, thanks for your work. I am trying to reproduce the results of MiniGrid-KeyCorridorS3R3. I use the following command
```
OMP_NUM_THREADS=1 python -m monobeast.minigrid.monobeast_amigo --env MiniGrid-KeyCorridorS3R3 \
--num_actors 40 --modify --generator_batch_size 150 --generator_entropy_cost .05 \
--generator_reward_negative -.3 --disable_checkpoint --modify \
--savedir ./experimentMinigrid
```
However, the `mean_episode_reward` is 0.0 after 2x 10^7 frames. In the paper, the agent should achieve around 0.9 within 1 x 10^7 frames. Am I doing something wrong?

I observe that some logs like `gen_rewards` and `generator_baseline_loss` are `None` in most of the logs. They will only have some values occasionally. Is it normal?

I install the dependencies with `pip install -r requirements.txt`. The installed dependencies are as follows.
```
Package      Version
------------ ----------
certifi      2020.4.5.2
cloudpickle  1.3.0
future       0.18.2
gitdb        4.0.5
GitPython    3.1.3
gym          0.17.2
gym-minigrid 1.0.1
numpy        1.18.5
pip          20.1.1
pyglet       1.5.0
scipy        1.4.1
setuptools   46.4.0
smmap        3.0.4
torch        1.5.1
wheel        0.34.2
```

Thanks!

----------------------------------------------------
Update

I observe the algorithm starts to converge on MiniGrid-KeyCorridorS3R3 after about 7 x 10 ^7 frames. However, it should converge much earlier.

**Comment by CrazySssst** [comment]
do you solve this problem? I meet same issue when run RIDE agent(https://github.com/facebookresearch/impact-driven-exploration)

**Comment by egrefen** [comment]
Pinging @rraileanu RE: issue with RIDE, and @ACampero for issues with AMIGo.

**Comment by rraileanu** [comment]
What commands are you using to train agents? It is important to set same number of --total_frames as shown in the paper. We use a learning rate scheduler which depends on the number of training steps. This can lead to different learning dynamics depending on args.total_frames. 

For training RIDE on MiniGrid-KeyCorridorS3R3: 
OMP_NUM_THREADS=1 python main.py --model ride --env MiniGrid-KeyCorridor-S3-R3-v0 --total_frames 30000000 --intrinsic_reward_coef 0.1 --entropy_cost 0.0005

Commands for training RIDE on other MiniGrid tasks are in the RIDE repo. Let me know if you are still having issues with reproducing the results. 

> do you solve this problem? I meet same issue when run RIDE agent(https://github.com/facebookresearch/impact-driven-exploration)

**Comment by ACampero** [comment]
For AMIGo:
- As @rraileanu  noted, set --total_frames as in the papers (default was previously set at different value): 
     6e8 for KeyCorridorS4R3-v0, KeyCorridorS5R3-v0, ObstructedMaze-2Dlhb-v0, ObstructedMaze-1Q-v0
     3e7 for KeyCorridorS3R3 and ObstructedMaze-1Dl-v0 
- hyperparameter --generator_threshold default has been slightly modified too (now set at -0.5)

This should now work, thanks for pointing it out.

**Comment by daochenzha** [comment]
@ACampero @rraileanu Thank you very much for the reply and the explanations!

**Comment by allepalma** [comment]
Hi, 

First, thank you for the amazing work you produced.

I know a long time has passed ever since this issue was posted, but I find myself in the same situation described in it. I am trying to train the agent on KeyCorridorS3R3-v0. My amigo environment is set up exactly as suggested by the original GitHub repository. I launch the code from the command prompt exactly as advised (with 3e7 as value for --num_actors as well).

What I am observing is that the value of t* (mean_episode_return) in the log messages indeed goes up, meaning that the student policy gets better at reaching the goals proposed by the teacher. However, the mean episode reward remains 0 all the time, and the mean_episode_steps goes up as well, meaning the adversarial training is not working properly.

I tried both with torch 1.5 and with an updated script version for torch 1.9 but, in both cases, I get no convergence at all when running the script as suggested. I have also hypothesized that the problem might be due to the uncertainty of training in a procedural environment. However, judging from the high confidence of convergence reported in the article, this should not be the case.

Therefore, I was wondering if @daochenzha was somehow able to reproduce the result (even increasing the number of frames to train on) and if @ACampero has any tips to get around the issue.

Thank you very much!

**Comment by ACampero** [comment]
Hey @allepalma this is strange. Note that --num__actors should be 40, not 3e7. 
I assume you meant --time_frames? Could you try with other values for --time_frames, and could you try another environment too?

**Comment by allepalma** [comment]
> Hey @allepalma this is strange. Note that --num__actors should be 40, not 3e7. I assume you meant --time_frames? Could you try with other values for --time_frames, and could you try another environment too?

Hi Andreas, thank you very much for your reply. Indeed, I meant --time_frames and not --num_actors (which I set to 40). Indeed the model ends up learning on medium tasks but I had to increase the used frames to 100M. After about 40 million frames it indeed takes off. As a reference, this is what it looks like.


![reward_kc](https://user-images.githubusercontent.com/65178831/138068209-dce75dd7-c800-40d5-b9d5-128e8156bbaf.png)
